code/datums/gamemode/role/vampire_role.dm

```diff
@@ -300,6 +300,9 @@
 			O.status &= ~ORGAN_BLEEDING
 	nullified = max(0, nullified - 1)
 
+	if (locate(/datum/power/vampire/mature) in current_powers)
+		H.dark_plane.alphas["vampire_vision"] = 255
```

Comment:
that's gonna be reset by human/update_perception()

A good fix would be to have a role/update_perception(), fit that in the vampire subtype, and call the roles' update_perception() in human/update_perception() after the reset.

Reply:
Vampires get that again updated on their `Life()` proc but yeah